File: EconRentCarWeb/EconRentCar.DataModel/TipoCombustible.cs
using EconRentCar.Core;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EconRentCar.DataModel
{
    public class TipoCombustible : IEntityBase
    {
        public TipoCombustible()
        {
            Vehiculos = new HashSet<Vehiculo>();
        }
        public Guid Id { get; set; }
        [Required]
        public string Nombre { get; set; }
        public bool Activo { get; set; }

        public virtual ICollection<Vehiculo> Vehiculos { get; set; }
    }
}

File: EconRentCarWeb/EconRentCarClient/src/app/shared/models/tipo-combustible.ts
import { BaseData } from '../dynamic-crud/models';
import { Vehiculo } from './vehiculo';

export class TipoCombustible implements BaseData {
    public Id = '00000000-0000-0000-0000-000000000000';
    public Nombre = '';
    public Activo = false;
    public Vehiculos: Vehiculo[];
}
